packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -532,7 +532,7 @@ export class Stack extends Construct implements ITaggable {
       params = new DockerImageAssetParameters(this.assetParameters, asset.sourceHash);
 
       const metadata: cxapi.ContainerImageAssetMetadataEntry = {
-        id: this.node.uniqueId,
+        id: asset.id,
```

Comment:
how about using `asset.directoryName` instead?

Reply:
`asset.directoryName` referes to the staging directory when assets are staged, this is equivalent to using `asset.sourceHash` which will recreate a repository with each change.

Reply:
Good point. damn!